### PR TITLE
Support tags on transformations

### PR DESCRIPTION
**Pull request by emelvarol-cognite**
## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [4.6.1] - 2022-09-28
```
Comment (silvavelosa) [inline CHANGELOG.md:17]: 4.7.0, since there is new features

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ Changes are grouped as follows
...
+### Changed
```
Comment (silvavelosa) [inline CHANGELOG.md:18]: Added

File: cognite/client/data_classes/transformations/__init__.py [hunk 3]
```diff
@@ -114,6 +116,7 @@ def __init__(
         destination_nonce: Optional[NonceCredentials] = None,
         source_session: Optional[SessionDetails] = None,
         destination_session: Optional[SessionDetails] = None,
+        tags: Optional[str] = None,
```
Comment (silvavelosa) [inline cognite/client/data_classes/transformations/__init__.py:119]:
```suggestion
        tags: Optional[List[str]] = None,
```

File: cognite/client/data_classes/transformations/__init__.py [hunk 4]
```diff
@@ -381,6 +400,33 @@ class TransformationList(CogniteResourceList):
...
+    """
+
+    def __init__(self, contains_any: List[str] = None, cognite_client: "CogniteClient" = None):
+        self.contains_any = contains_any
+        self._cognite_client = cast("CogniteClient", cognite_client)
+
+    def dump(self, camel_case: bool = False) -> Dict[str, Any]:
+        dump_key = lambda key: key if not camel_case else utils._auxiliary.to_camel_case(key)
+        wrap = lambda values: None if values is None else [{"externalId": value} for value in values]
+        return {dump_key(key): wrap(value) for key, value in self.items()}
```
Comment (silvavelosa) [inline cognite/client/data_classes/transformations/__init__.py:425]: don't understand this 😅 .... do we need to wrap in `{"externalId": _}`?

**Review by silvavelosa — COMMENTED** [review]
Awesome! 🚀 
just some small comments, but LGTM ✔️ 
(we need @cognitedata/starbase approval anyway, to be able to skip annotations tests)

File: cognite/client/data_classes/transformations/__init__.py (hunk 4 above)
Comment (emelvarol-cognite) [reply cognite/client/data_classes/transformations/__init__.py]: it was copy paste, fixed it 👍

File: cognite/client/data_classes/transformations/__init__.py [hunk 5]
```diff
@@ -381,6 +401,34 @@ class TransformationList(CogniteResourceList):
     _RESOURCE = Transformation
 
 
+class TagsFilter:
+    @abstractmethod
+    def dump(self) -> Dict[str, Any]:
+        ...
```
Comment (emelvarol-cognite) [inline cognite/client/data_classes/transformations/__init__.py:407]: this will possibly be extended with `containsAll` later.

**Review by silvavelosa — APPROVED** [review]
Awesome! 🚀

File: cognite/client/data_classes/transformations/__init__.py (hunk 5 above)
Comment (silvavelosa) [reply cognite/client/data_classes/transformations/__init__.py:407]: Nice approach!

**Pull request merged**